penty/pentypes/builtins.py

```diff
@@ -291,6 +291,136 @@ def float_str(self_ty):
 for slot in ('add', 'divmod', 'floordiv', 'mod', 'mul', 'pow', 'sub', 'truediv'):
     _float_attrs['__r{}__'.format(slot)] = _float_attrs['__{}__'.format(slot)]
 
+##
+#
+
+def complex_abs(self_ty):
+    if not issubclass(self_ty, complex):
+        raise TypeError
+    return complex
+
+def complex_bool(self_ty):
+    if not issubclass(self_ty, complex):
+        raise TypeError
+    return bool
+
+def complex_divmod(self_ty, other_ty):
+    if not issubclass(self_ty, complex):
+        raise TypeError
+    raise TypeError
```

Comment:
You raise `TypeError` whatever happens. The first check is most probably useless.

Reply:
Yeah, that's just forlater, when an actual error message will be used :-)